File: lc_how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        ans = [0] * len(nums)
        for i in range(len(nums)):
            ansI = 0
            for j in range(0,len(nums)):
                if j != i and nums[j] < nums[i]:
                    ansI += 1
            ans[i] = ansI
        return ans

File: lc_how-many-numbers-are-smaller-than-the-current-number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
// O(n^2) solution
var smallerNumbersThanCurrent = function (nums) {
  const answer = new Array(nums.length).fill(0);
  for (let i = 0; i < nums.length; i++) {
    answer[i] = nums.filter((el, idx) => el < nums[i] && idx != i).length;
  }
  return answer;
};
// Better
var smallerNumbersThanCurrent2 = function (nums) {
  const arr = [...nums].sort((a, b) => a - b);
  return nums.map((el) => arr.indexOf(el));
};
